workflow/scripts/validate_fasta.py

```diff
@@ -0,0 +1,28 @@
+from Bio import SeqIO
+
+input_fasta = snakemake.input["fasta"]
+output_fasta = snakemake.output["fasta"]
+log = snakemake.log
```

Comment:
Maybe you have a particular reason you use this contrast. If so, I'd be curious to know. Maybe this has some advantage over our standard solution, which is this boilerplate code instead:
```suggestion
import sys
sys.stderr = open(snakemake.log[0], "w", buffering=1)
```

We have this documented here, at the moment:
https://koesterlab.github.io/data-science-for-bioinfo/workflows/snakemake.html#logging-1

And with this, all the calls in `validate_fasta()` would simply become `sys.stderr.write("\n".join(summary)` and similar.

Reply:
i was not aware of this log redirection! well good we'll have it in the template so other users can copy that.